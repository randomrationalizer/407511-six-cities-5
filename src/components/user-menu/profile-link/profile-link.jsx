import React from "react";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import PropTypes from "prop-types";
import {AppRoute} from "../../../const";

const avatarSize = {
  width: 54,
  height: 54
};

const ProfileLink = (props) => {
  const {userInfo} = props;

  return (
    <Link className="header__nav-link header__nav-link--profile" to={AppRoute.FAVORITES}>
      <div className="header__avatar-wrapper user__avatar-wrapper">
        <img
          className="user__avatar"
          src={userInfo.avatar_url}
          width={avatarSize.width}
          height={avatarSize.height}
          alt="User avatar"
        />
      </div>
      <span className="header__user-name user__name">{userInfo.email}</span>
    </Link>
  );
};

ProfileLink.propTypes = {
  userInfo: PropTypes.object.isRequired
};

const mapStateToProps = ({USER}) => ({
  userInfo: USER.userInfo
});

export {ProfileLink};
export default connect(mapStateToProps, null)(ProfileLink);
