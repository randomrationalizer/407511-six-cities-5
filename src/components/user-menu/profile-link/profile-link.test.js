import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {ProfileLink} from "./profile-link";
import mockUserInfo from "../../../mocks/test-data/user-info";


test(`ProfileLink component renders correctly`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <ProfileLink
            userInfo={mockUserInfo}
          />
        </BrowserRouter>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
