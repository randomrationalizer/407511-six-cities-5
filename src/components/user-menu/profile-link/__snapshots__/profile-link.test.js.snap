// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileLink component renders correctly 1`] = `
<a
  className="header__nav-link header__nav-link--profile"
  href="/favorite"
  onClick={[Function]}
>
  <div
    className="header__avatar-wrapper user__avatar-wrapper"
  >
    <img
      alt="User avatar"
      className="user__avatar"
      height={54}
      src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg"
      width={54}
    />
  </div>
  <span
    className="header__user-name user__name"
  >
    Oliver.conner@gmail.com
  </span>
</a>
`;
